src/zjs_net.c

```diff
@@ -259,13 +259,12 @@ static void tcp_received(struct net_context *context,
 {
     sock_handle_t *handle = (sock_handle_t *)user_data;
 
-    if (handle) {
+    if (handle && buf) {
         start_socket_timeout(handle, handle->timeout);
 
         uint32_t len = net_nbuf_appdatalen(buf);
         uint8_t *data = net_nbuf_appdata(buf);
 
-        DBG_PRINT("received data, context=%p, data=%p, len=%u\n", context, data, len);
         // TODO: Probably not the right way to check if a socket is closed
```

Comment:
Ok that makes sense. So what does if mean if buf is valid, but data == NULL and len == 0? When I wrote this and was testing, I would always get that case when I closed the other end of the connection.